pkg/cli/image/manifest/dockercredentials/auth_resolver.go

```diff
@@ -0,0 +1,156 @@
+package dockercredentials
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/containers/image/v5/docker/reference"
+
+	"github.com/openshift/oc/pkg/helpers/image/credentialprovider"
+)
+
+type AuthResolver struct {
+	authConfigs credentialprovider.DockerConfig
+}
+
+// NewAuthResolver creates a new auth resolver that loads authFilePath file
+// (defaults to a docker locations) to find a valid
+// authentication for registry targets.
+func NewAuthResolver(authFilePath string) (*AuthResolver, error) {
+	var cfg credentialprovider.DockerConfig
+	var err error
+
+	if authFilePath != "" {
+		cfg, err = credentialprovider.ReadSpecificDockerConfigJSONFile(authFilePath)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		cfg = defaultClientDockerConfig()
+	}
+
+	return &AuthResolver{
+		authConfigs: cfg,
+	}, nil
+}
+
+// original: https://github.com/containers/image/blob/main/pkg/docker/config/config.go
+// findAuthentication looks for auth of registry in path. If ref is
+// not nil, then it will be taken into account when looking up the
+// authentication credentials.
+func (r *AuthResolver) findAuthentication(ref reference.Named, registry string) (credentialprovider.DockerConfigEntry, error) {
```

Comment:
It seems this function is always called with `ref, reference.Domain(ref)`, can `registry string` be removed?